user/serializers.py

```diff
@@ -1 +1,36 @@
-# write your code here
+from django.contrib.auth import get_user_model
+from rest_framework import serializers
+
+from user.models import User
+
+
+class UserSerializer(serializers.ModelSerializer):
+    class Meta:
+        model = get_user_model()
+        fields = [
+            "id",
+            "username",
+            "password",
+            "email",
+            "is_staff"
+        ]
+        read_only_fields = [
+            "id",
+            "is_staff"
+        ]
+        extra_kwargs = {
+            "password": {
+                "write_only": True,
+                "min_length": 5
+            }
+        }
+
+    def create(self, validated_data):
+        return get_user_model().objects.create_user(**validated_data)
+
+    def update(self, instance, validated_data):
+        password = validated_data.pop("password", None)
+        if password:
+            instance.set_password(password)
+        super().update(instance, validated_data)
+        return instance
```

Comment:
You should call super().update() before setting the password to make sure the remaining validated data is processed correctly.